### DISCUSSION TITLE
Where did @json and @lang go?

### DISCUSSION THREAD

**Discussion by raveren** [laravel/framework#44248] [Q&A]
I noticed `@lang` and `@json` are no longer mentioned in the latest [blade documentation](https://laravel.com/docs/9.x/blade)?

There is zero information about this change in the internet, including the 9.x upgrade guide.

Here's the commit where `@json` bites the dust:

https://github.com/laravel/docs/commit/73165619680fdb6fe44170a70342a58dadd1455b by @taylorotwell 

## Question: are these blade directives silently being deprecated?

**Comment by ubised** [comment]
The commit you have linked to shows what it has been replaced with.

I guess you are asking why the change. For me it makes more sense to have the `Js` class, following the same convention as `Arr` and `Str`. It is then available wherever you need it, not just blade, and can be changed and extended in due course, I would expect.

If you want to keep `@json` after the upgrade, you have the option of creating it as a custom blade directive https://laravel.com/docs/9.x/blade#extending-blade

**Reply by raveren** [reply]
Thanks for your personal opinion on the matter, but what I think the community needs is an official answer on what's going on.

**Reply by henzeb** [reply]
Then get the attention of @driesvints or @taylorotwell. But I too think this is the reason.

**Reply by raveren** [reply]
Huh I don't know why taylorotwell was not tagged successcully in the OP, strange.

**Comment by brandonburkett** [comment]
I ran into this today when checking the docs.  Is there an end of life for the `@json` directive before we need to migrate to the Js class?

**Reply by raveren** [reply]
it is trivial to just write your own `@json()` custom blade directive if it does get EOL

**Reply by brandonburkett** [reply]
No biggie and I get it... just didn't see it in the release / migration notes when we upgraded to Laravel 9 so have been using `@json` as if it was still the standard.  It wasn't until I checked the docs and noticed it was missing that I stumbled on this post.

Looks like it is in "old way, but still supported" mode for now.  Thanks!

**Comment by browner12** [comment]
This was my proposal to try and deal with this issue:

https://github.com/laravel/docs/pull/8590
